[WIP] Add gemma4 drafter model support (#2240)
NVIDIA-NeMo/Automodel: docs/model-coverage/vlm/google/gemma4.md

Reviewer: We try to avoid latinisms in text, such as "via"

```suggestion
[Gemma 4](https://ai.google.dev/gemma) is Google's next-generation multimodal Gemma family, supporting image-text inputs with a Mixture-of-Experts (MoE) language backbone at larger scales. NeMo AutoModel replaces the HF-native dense matmul over experts with the NeMo `GroupedExperts` backend, enabling Expert Parallelism (EP) through the standard MoE parallelizer.
```

Reviewer: ```suggestion
**2. Navigate to the AutoModel directory** (where the recipes are):
```

## Available Models

- **Gemma 4 E2B IT** (VL, dense, kv-shared layers)
- **Gemma 4 E4B IT** (VL, dense, kv-shared layers), **Gemma 4 E4B IT Assistant** (Assistant/drafter model for MTP)
- **Gemma 4 31B IT** (VL, dense)
- **Gemma 4 26B-A4B IT** (VL, MoE)

## Architecture

- `Gemma4ForConditionalGeneration`

> - `Gemma4AssistantForCausalLM` (MTP drafter / assistant for speculative decoding; co-trainable with the target Gemma 4 base via `Gemma4WithDrafter`)

Reviewer: ```suggestion
- `Gemma4AssistantForCausalLM` (MTP drafter / assistant head for speculative decoding; co-trainable with the target Gemma 4 base using `Gemma4WithDrafter`)
```
